### PR TITLE
[ASCII-145] Move apm status to `pkg/status/apm` 

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Following how the status package fetches the process agent status, we are changing how we fetch the APM status.

Rather than doing an extra call from the CLI `agent status` (Client), we fetch it directly from the agent backend (`pkg/status`)

That means that CLI no longer has to worry about fetching the APM status or any other status for any other agent process; the backend takes care of fetching everything for it and making the CLI code simpler.

As an added bonus, the GUI was using AJAX to replicate the CLI behaviour; since the GUI is using the `pkg/status` to fetch that information, we no longer have to maintain that JS code for the GUI. 🔥 

#### Context 
Here is the PR that introduced the trace agent status https://github.com/DataDog/datadog-agent/pull/5235
Here is the PR that added the process agent status https://github.com/DataDog/datadog-agent/pull/10886

#### Results

Here is a screenshot of the GUI using the current branch

Screenshot 2023-11-23 at 17 30 54

Here is the output of APM section of `agent status`

```
=========
APM Agent
=========
  Status: Running
  Pid: 81185
  Uptime: 42 seconds
  Mem alloc: 11,599,808 bytes
  Hostname: COMP-VQHPF4W6GY
  Receiver: 0.0.0.0:8126
  Endpoints:
    https://trace.agent.datadoghq.eu

  Receiver (previous minute)
  ==========================
    No traces received in the previous minute.


  Writer (previous minute)
  ========================
    Traces: 0 payloads, 0 traces, 0 events, 0 bytes
    Stats: 0 payloads, 0 stats buckets, 0 bytes

```

### Describe how to test/QA your changes

Make sure the trace agent and the agent is running.

To enable the trace agent, add this to your configuration:

```
apm_config:
  enabled: true
  apm_non_local_traffic: true
```

Once they are running we should validate that:
- [ ] `agent status` returns the correct information
- [ ] That the APM information is displayed correctly in the GUI. To launch the GUI run `agent launch-gui`

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/agent/gui/views/templates/generalStatus.tmpl [hunk 1]
```diff
@@ -479,9 +479,61 @@
     </span>
   </div>
 
-  <div class="stat" id="apmStats">
+  <div class="stat">
     <span class="stat_title">APM</span>
-    <span class="stat_data">Loading...</span>
+    <span class="stat_data">
```
Comment (truthbk) [inline cmd/agent/gui/views/templates/generalStatus.tmpl:484]: 🍰 way better

**Review by truthbk — APPROVED** [review]
Looks good! 🚀

File: pkg/status/apm/apm.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// httpClients should be reused instead of created as needed. They keep cached TCP connections
+// that may leak otherwise
+var (
+	httpClient     *http.Client
+	clientInitOnce sync.Once
+)
+
+func getHTTPClient() *http.Client {
+	clientInitOnce.Do(func() {
+		httpClient = apiutil.GetClient(false)
```
[the remark this answers is not in the document]
Comment (GustavoCaso) [reply pkg/status/apm/apm.go:29]: The `apiutil.GetClient` does not allow to provide a timeout 😢 

I mimic the process agent code for fetching the information. They don't seem to be using any timeout, and there isn't any issue.  

https://github.com/DataDog/datadog-agent/blob/df1273705b4e05bb5b1257202ac6dae0c06caffa/pkg/status/status_process_agent.go#L47

**Review by katiehockman — APPROVED** [review]

**Pull request merged**